src/libs/ApiUtils.js

```diff
@@ -0,0 +1,85 @@
+import lodashGet from 'lodash/get';
+import Onyx from 'react-native-onyx';
+import ONYXKEYS from '../ONYXKEYS';
+import CONFIG from '../CONFIG';
+import CONST from '../CONST';
+import * as Environment from './Environment/Environment';
+
+// To avoid rebuilding native apps, native apps use production config for both staging and prod
+// We use the async environment check because it works on all platforms
+let ENV_NAME = CONST.ENVIRONMENT.PRODUCTION;
+Environment.getEnvironment()
+    .then((envName) => {
+        ENV_NAME = envName;
+    });
+
+let stagingServerToggleState = false;
+Onyx.connect({
+    key: ONYXKEYS.USER,
+    callback: (val) => {
+        const defaultValue = ENV_NAME === CONST.ENVIRONMENT.STAGING;
+        stagingServerToggleState = lodashGet(val, 'shouldUseStagingServer', defaultValue);
+    },
+});
+
+/**
+ * Helper method used to decide which API endpoint to call
+ * Non PROD environments allow API switching via the {@link stagingServerToggleState}
+ *
+ * @returns {Boolean}
+ */
+function canUseStagingServer() {
```

Comment:
Given that both dev and staging **_can_** use staging server, I think we should name this method shouldUseStagingServer 
```suggestion
function shouldUseStagingServer() {
```

Reply:
It was initially called `shouldUseStagingServer` but an update was requested: https://github.com/Expensify/App/pull/15178#discussion_r1128487808